src/apm/helper.go

```diff
@@ -214,17 +214,7 @@ func (i *baseInjector) validate(inst v1alpha2.Instrumentation) error {
 	return nil
 }
 
-func (i *baseInjector) injectNewrelicConfig(ctx context.Context, resource v1alpha2.Resource, ns corev1.Namespace, pod corev1.Pod, index int, licenseKeySecretName string) corev1.Pod {
-	container := &pod.Spec.Containers[index]
-
-	if idx := getIndexOfEnv(container.Env, EnvNewRelicAppName); idx == -1 {
-		//@todo: how can we do this if multiple injectors need this?
-		resourceMap := i.createResourceMap(ctx, resource, ns, pod, index)
-		container.Env = append(container.Env, corev1.EnvVar{
-			Name:  EnvNewRelicAppName,
-			Value: chooseServiceName(pod, resourceMap, index),
-		})
-	}
+func (i *baseInjector) injectNewrelicLicenseKeyIntoContainer(container corev1.Container, licenseKeySecretName string) corev1.Container {
```

Comment:
Can we change this function to something like this
func (i *baseInjector) injectNewrelicLicenseKeyIntoContainer(container * corev1.Container, licenseKeySecretName string) bool {
.....
}

Looks like we are changing the passed the container so why not pass this as a pointer and return bool if this was actually done for tracking upstream.

Reply:
I'll leave that up to  the agents.  This uses copies rather than pointers just like the majority of the code in the helpers.  It's also not beneficial to return true/false, as it could only really be used for logging, and even then the logs may just may make it more confusing.